src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationServiceImpl service = new RegistrationServiceImpl();
+    private StorageDaoImpl storage;
+
+    @BeforeEach
+    void setUp() {
+        storage = new StorageDaoImpl();
+        service = new RegistrationServiceImpl();
```

Comment:
The service instance is being initialized twice: once at declaration (line 14) and again in the setUp method (line 20). You should remove the initialization at declaration to avoid redundancy since setUp method with @BeforeEach will initialize it before each test.